packages/portfolio-contract/src/durable-queue.exo.ts

```diff
@@ -0,0 +1,129 @@
+/**
+ * @file Durable queue helper backed by a durable map and index pointers.
+ */
+import { Fail } from '@endo/errors';
+import { M } from '@endo/patterns';
+import type { MapStore, StoreOptions } from '@agoric/store';
+import type { Zone } from '@agoric/zone';
+
+export type DurableQueueState<V> = {
+  readonly entries: MapStore<bigint, V>;
+  head: bigint;
+  tail: bigint;
+};
+
+export type DurableQueue<V> = ReturnType<
+  ReturnType<typeof prepareDurableQueue<V>>
+>;
+
+export const prepareDurableQueue = <V>(
+  zone: Zone,
+  queueName: string,
+  options: StoreOptions = {},
+) => {
+  const collections = zone.detached();
+
+  const indicesShape = M.splitRecord(
+    {
+      head: M.bigint(),
+      tail: M.bigint(),
+    },
+    {},
+  );
+
+  const entryShape = M.splitRecord(
+    {
+      index: M.bigint(),
+      value: options.valueShape || M.any(),
+    },
+    {},
+  );
+
+  const apiShape = options.valueShape || M.any();
```

Comment:
`apiShape` seems like it should be the shape of the API; i.e. all the methods.


```suggestion
  const itemShape = options.valueShape || M.any();
```

or

```suggestion
  const valueArgShape = options.valueShape || M.any();
```